CriteoPublisherSdk/Sources/Logging/CR_RemoteLogRecordSerializer.m

```diff
@@ -0,0 +1,67 @@
+//
+//  CR_RemoteLogRecordSerializer.m
+//  CriteoPublisherSdk
+//
+//  Copyright © 2018-2021 Criteo. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "CR_RemoteLogRecordSerializer.h"
+
+@implementation CR_RemoteLogRecordSerializer
+
+- (NSArray<NSDictionary<NSString *, NSObject *> *> *)bodyForRemoteLogRecords:
+    (NSArray<CR_RemoteLogRecord *> *)records {
+  NSMutableArray *formattedArray = [NSMutableArray arrayWithCapacity:records.count];
+  for (CR_RemoteLogRecord *record in records) {
+    NSDictionary<NSString *, NSObject *> *dictionary = [self dictionaryForRemoteLogRecord:record];
+    [formattedArray addObject:dictionary];
+  }
+  return formattedArray;
+}
+
+- (NSDictionary<NSString *, NSObject *> *)dictionaryForRemoteLogRecord:
+    (CR_RemoteLogRecord *)record {
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "OCLegacyObjCLiteralInspection"
```

Comment:
NIT: If this very specific to AppCode I would suggest removing it as the comment is quite explicit. Maybe also add a warning about adding trailing entries using this constructor or maybe use `dictionaryWithObjects:forKeys:` instead.

Reply:
Ok let me remove the `OCLegacyObjCLiteralInspection`.

And you're right, `dictionaryWithObjects:forKeys:` seems better to use when handling `nil` values.

Reply:
In fact `dictionaryWithObjects:forKeys:` needs an array. For the same reason, `@[]` notation can't be used. So I need to use `+ (instancetype)arrayWithObjects:(ObjectType)firstObj, ... NS_REQUIRES_NIL_TERMINATION;` which has the same issue with `nil`.
So instead, I will use a mutable dictionary if it's ok for you.

Reply:
As you wish, this is just NIT, the former version is fine as well :)